.github/workflows/analyzer.yml

```diff
@@ -0,0 +1,32 @@
+name: Analyze Code
+
+on:
+  push:
+    branches: dev
+    paths: 
+      - "**.dart"
+      - ".github/workflows/analyze.yml"
+  pull_request:
+    branches: [ "main", "dev" ]
```

Comment:
Wouldn't it be relevant for all branches?

Reply:
This will cause the analyser to run on unintended branches like `compose-dev`

Reply:
No because the workflow isn't in that branch and also even if it is, there's the path check

Reply:
Limit to only supported branch is safer than not specifying branch at all; in the future we might introduce analyser for Compose branch or something unrelated to Flutter.

Reply:
This workflow won't run in the compose-dev branches because the workflow file isn't in the branch

Reply:
Yep, but in the future we might introduce Compose analyser which can cause problems with running Dart-specific command.